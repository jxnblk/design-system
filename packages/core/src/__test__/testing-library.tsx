import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
