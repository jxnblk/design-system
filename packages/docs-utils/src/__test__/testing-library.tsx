import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
