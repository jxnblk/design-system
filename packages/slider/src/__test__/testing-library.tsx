import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
