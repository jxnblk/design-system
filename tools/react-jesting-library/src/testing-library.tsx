import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
