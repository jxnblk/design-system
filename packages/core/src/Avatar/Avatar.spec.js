import React from 'react'
import { render } from 'testing-library'

import { Avatar } from '..'

const elonJPG = 'https://pbs.twimg.com/media/DwSta0wUcAAQQR9.jpg'

describe('Avatar', () => {
  test('renders default', () => {
    const json = rendererCreateWithTheme(<Avatar />)
    expect(json).toMatchSnapshot()
  })

  test('renders initials', () => {
    const json = rendererCreateWithTheme(<Avatar initials='WS' />)
    expect(json).toMatchSnapshot()
  })

  test('renders mr elon', () => {
    const { asFragment } = render(
      <Avatar
        title='Not Elon Musk'
        subtitle='totally.not.elon@musk.com'
        src={elonJPG}
        altText='papa-elon'
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
