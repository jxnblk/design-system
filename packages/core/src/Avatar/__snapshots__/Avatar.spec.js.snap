// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 24px;
}

.c3 {
  outline: none;
}

.c1 {
  padding: 8px;
  background-color: #007aff;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-size: 40px;
  background-image: url();
}

<div
  className="c0"
>
  <div
    className="c1"
    color="primary"
  >
    <svg
      aria-hidden={true}
      className="c2 c3"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 12c2.2 0 4-1.8 4-4s-1.8-4-4-4-4 1.8-4 4 1.8 4 4 4zm0 2c-2.7 0-8 1.3-8 4v2h16v-2c0-2.7-5.3-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`Avatar renders initials 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  font-size: 14px;
}

.c1 {
  padding: 8px;
  background-color: #007aff;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-size: 40px;
  background-image: url();
}

<div
  className="c0"
>
  <div
    className="c1"
    color="primary"
  >
    <div
      className="c2"
      fontSize={1}
    >
      WS
    </div>
  </div>
</div>
`;

exports[`Avatar renders mr elon 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  margin: 0;
}

.c5 {
  color: #4f6f8f;
  font-size: 12px;
}

.c2 {
  padding: 8px;
  background-color: #007aff;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-size: 40px;
  background-image: url(https://pbs.twimg.com/media/DwSta0wUcAAQQR9.jpg);
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        color="primary"
        src="https://pbs.twimg.com/media/DwSta0wUcAAQQR9.jpg"
      />
      <div
        class="c3"
      >
        <h5
          class="c4"
        >
          Not Elon Musk
        </h5>
        <div
          class="c5"
          color="text.light"
          font-size="0"
        >
          totally.not.elon@musk.com
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
